browser/net/brave_ads_status_header_network_delegate_helper_unittest.cc

```diff
@@ -0,0 +1,142 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
```

Comment:
I was not sure by reading the code so I'll just ask; do we have test-cases here to make sure the header is added for non-`main_frame` requests as well here?

Reply:
Thanks, there are browser tests which checks that all requests has the header and explicitly checks that the header is added for a fetch request. But I think it's a good idea to add checks for resource type here, I will add these checks